### DISCUSSION TITLE
Import data from MoneyManager app

### DISCUSSION THREAD

**Discussion by mrvalterhugo** [firefly-iii/firefly-iii#6895] [Firefly III]
Hi, I am wondering if anyone has successfully imported data from app MoneyManager.
The data is exported in excel format. Which I have them converted to CSV. I have 5 years of transactions. I tested importing a subset of data but it took me a while to configure all categories. Before going all in, I wanted to check if there is an easier way of doing it of if anyone has done this before. Thank You.

**Comment by Maks-s** [comment]
For anyone wondering, I did a successful import from MoneyManager by Realbyte to Firefly 3, so here's a quick tutorial

First, you need to make a backup using "More > Backup > Backup/restore on device", and you'll have to transfer the file to your computer. You should be able to use "Export backup files to e-mail" to transfer the file more easily, but I haven't tested this functionality

The backup file should look something like `MM(01-01-2024-412445).mmbak`. This is actually a SQLite file, and you'll have to open it using `sqlite3`, `SQLite Browser`, or something else which can export as csv.

After opening the backup file, you need to run the following SQL code and export the result as a `csv` with headers. I recommend using a comma (`,`) as a separator, and double quotes (`"`) for fields (it worked for me).
```sql
SELECT
	asset.NIC_NAME AS account,
	-- This weird code handles Category w/ Subcategories
	(CASE
		WHEN tx.ctgUid = '-4' THEN 'Bal. Difference'
		WHEN parentCat.uid IS NULL THEN cat.name
		ELSE parentCat.name
	END) AS category,
	(CASE WHEN parentCat.uid IS NULL THEN
		NULL
	ELSE
		cat.name
	END) AS tags,
	assetReceiving.NIC_NAME AS toAccount,
	tx.ZCONTENT AS description,
	(CASE tx.DO_TYPE
		WHEN 0 THEN 'credit'
		WHEN 1 THEN 'debit'
		WHEN 3 THEN 'debit' -- Actually a transfer, but it's marked as Debit so Firefly imports it correctly
		WHEN 7 THEN 'credit' -- Modified Bal positive
		WHEN 8 THEN 'debit' -- Modified Bal negative
	END) AS indicator,
	tx.ZMONEY AS amount,
	curr.ISO AS foreignCurrencyISO,
	tx.IN_ZMONEY AS foreignAmount,
	tx.ZDATA AS note,
	-- Firefly supports Unix Timestamp in seconds only, but MM has milliseconds
	CAST(CAST(tx.ZDATE AS REAL) / 1000 AS INT) AS unixDate
FROM [INOUTCOME] tx
LEFT JOIN [CURRENCY] curr ON curr.uid = tx.currencyUid
LEFT JOIN [ASSETS] asset ON asset.uid = tx.assetUid
LEFT JOIN [ASSETS] assetReceiving ON assetReceiving.uid = tx.toAssetUid
LEFT JOIN [ZCATEGORY] cat ON cat.uid = tx.ctgUid
LEFT JOIN [ZCATEGORY] parentCat ON parentCat.uid = cat.pUid
WHERE
	(tx.IS_DEL = 0 OR tx.IS_DEL IS NULL) -- We don't care about deleted transactions
	AND tx.DO_TYPE <> 4 -- Each transfert creates 2 transactions. We only use one, DO_TYPE 3 (src -> dest)
```

Do you have the `.csv` file ? Congratulations, the hardest part is behind you ! You can fire up the Data Importer, [grab my config file](https://github.com/user-attachments/files/17347593/config.json), make sure everything is ok, **check "Map Data" for the columns "account" and "toAccount"**, map every account or create new ones so the import won't break, and hit "Submit".

After waiting, every transaction should be imported in Firefly :slightly_smiling_face:
